lib/ansible/modules/remote_management/manageiq/manageiq_dynamic_resource_definition.py

```diff
@@ -0,0 +1,270 @@
+           provider: ManageIQ::Providers::CloudManager
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+
+- name: Edit a dynamic resource definition in ManageIQ
+  manageiq_dynamic_resource_definition:
+    name: 'my_edited_custom_definition'
+    state: 'present'
+    properties:
+         attributes:
+           group: string
+           engine: string
+           type: string
+           size: integer
+           region: string
+           username: string
+         associations:
+           service: Service
+           provider: ManageIQ::Providers::CloudManager
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+
+- name: Delete a dynamic resource definition in ManageIQ
+  manageiq_dynamic_resource_definition:
+    name: 'my_edited_custom_definition'
+    state: 'absent'
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.manageiq import ManageIQ, manageiq_argument_spec
+
+
+class ManageIQDynamicResourceDefinition(object):
+    """
+        Object to create dynamic resource definition operations in manageiq.
+    """
+
+    def __init__(self, manageiq):
+        self.manageiq = manageiq
+        self.module = self.manageiq.module
+        self.api_url = self.manageiq.api_url
+        self.client = self.manageiq.client
+
+    def dynamic_resource_definition(self, name):
+        """ Search for dynamic resource definition by name name.
+
+        Returns:
+            the dynamic resource definition, or None if not found.
+        """
+        return self.manageiq.find_collection_resource_by('generic_object_definitions', name=name)
+
+    def create_definition(self, dynamic_resource_definition, name, properties):
+        """ Creates the dynamic resource definition in manageiq.
+
+        Returns:
+            a dictionary with changed and msg.
+        """
+        url = '%s/generic_object_definitions' % (self.api_url)
+
+        resource = {
+            'name': name,
+            'properties': properties,
+        }
+
+        try:
+            result = self.client.post(url, action='create', resource=resource)
+        except Exception as e:
+            self.module.fail_json(
+                msg="failed to create dynamic resource definition %s: %s" % (name, str(e))
+            )
+
+        return dict(
+            changed=True,
+            msg="successfully created the dynamic resource definition %s: %s" % (name, result['results'])
+        )
+
+    def edit_definition(self, dynamic_resource_definition, name, properties):
+        """ Edit the dynamic resource definition in manageiq.
+
+        Returns:
+            a dictionary with the changed and msg.
```

Comment:
I think this is a typo - `changed` should be `changes`

Reply:
no, this refers to the standard `changed=True` / `changed=False` and `msg="..."` results returned by ansible modules.